Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -141,13 +141,20 @@ try {
                 if ($ghEnvironment) {
                     $branchPolicy = ($ghEnvironment.protection_rules | Where-Object { $_.type -eq "branch_policy" })
                     if ($branchPolicy) {
-                        Write-Host "GitHub Environment $envName has branch policies, getting branches from GitHub API"
-                        $branchesUrl = "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/environments/$([Uri]::EscapeDataString($envName))/deployment-branch-policies"
-                        Write-Host "Getting branches for $envName from GitHub API"
-                        $branches = @((InvokeWebRequest -Headers $headers -Uri $branchesUrl -ignoreErrors | ConvertFrom-Json).branch_policies | ForEach-Object { $_.name })
+                        $branches = @()
+                        if ($ghEnvironment.deployment_branch_policy.protected_branches) {
+                            Write-Host "GitHub Environment $envName only allows protected branches, getting protected branches from GitHub API"
+                            $branchesUrl = "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/branches"
+                            $branches = @((InvokeWebRequest -Headers $headers -Uri $branchesUrl -ignoreErrors | ConvertFrom-Json) | Where-Object { $_.protected } | ForEach-Object { $_.name })
```

Comment:
Few lines below `$branches` is overwritten if defined in the settings.
It is better to first check the settings and only query the branch policies if branches are not defined in the settings.

```
if (($settings.Contains($deployToName)) -and ($settings."$deployToName".Contains('Branches'))) {
    $branches = @($settings."$deployToName".Branches)
}
```